python_modules/dagster/dagster/_core/storage/migration/bigint_migration.py

```diff
@@ -0,0 +1,111 @@
+from typing import Callable
+
+import sqlalchemy as db
+
+from dagster._core.instance import DagsterInstance
+from dagster._core.storage.event_log import (
+    SqlEventLogStorage,
+    SqlEventLogStorageMetadata,
+    SqliteEventLogStorage,
+)
+from dagster._core.storage.runs import RunStorageSqlMetadata, SqliteRunStorage, SqlRunStorage
+from dagster._core.storage.schedules import (
+    ScheduleStorageSqlMetadata,
+    SqliteScheduleStorage,
+    SqlScheduleStorage,
+)
+
+
+def _has_integer_id_column(inspector, table):
+    type_by_col_name = {col["name"]: col["type"] for col in inspector.get_columns(table)}
+    id_col_type = type_by_col_name.get("id")
+    return id_col_type and str(id_col_type) == str(db.Integer())
+
+
+def _convert_int_to_bigint(conn, table, colname):
+    # Since we are not using alembic (which might mess up the versioning system), we need to
+    # manually handle the dialect differences for running the migration
+    dialect = db.inspect(conn).dialect.name
+    if dialect == "postgresql":
+        statement = db.text(f"ALTER TABLE {table} ALTER COLUMN {colname} TYPE BIGINT")
+    elif dialect == "mysql":
+        statement = db.text(f"ALTER TABLE {table} MODIFY COLUMN {colname} BIGINT")
+    else:
+        raise Exception(f"Unsupported dialect {dialect}")
+    conn.execute(statement)
+
+
+def _drop_foreign_key_constraint(conn, table, fk):
+    conn.execute(db.text(f"ALTER TABLE {table} DROP CONSTRAINT {fk['name']}"))
+
+
+def _add_foreign_key_constraint(conn, table, fk):
+    constrained_columns = ", ".join(fk["constrained_columns"])
+    referred_columns = ", ".join(fk["referred_columns"])
+    conn.execute(
+        db.text(
+            f"ALTER TABLE {table} ADD CONSTRAINT {fk['name']} FOREIGN KEY ({constrained_columns})"
+            f" REFERENCES {fk['referred_table']}({referred_columns}) ON DELETE CASCADE"
+        )
+    )
+
+
+def _migrate_storage(conn, metadata, print_fn):
+    inspector = db.inspect(conn)
+    storage_tables = [table for table in inspector.get_table_names() if table in metadata.tables]
+    to_migrate = set()
+    for table in storage_tables:
+        if _has_integer_id_column(inspector, table):
+            to_migrate.add(table)
+
+    dropped_fks = []
```

Comment:
To Alex's point, I think we might have trouble resuming from failure if a column gets migrated but the fk operations fail. The next time through, we won't have the table in `to_migrate` and we'll exit early.

Maybe we should just hardcode the list of tables? We know what they are and they aren't changing. Have each operation be a no-op if the table is already fully migrated.